src/openads-appnexus/AppNexusConnector.js

```diff
@@ -1,87 +1,91 @@
-   * @param callback
-   */
-  onEvent ({event, targetId, callback}) {
-    throw new Error('AppNexusConnector#onEvent must be implemented')
+  constructor ({source, member}) {
+    this._astClient = new AstClientImpl({
+      source: source,
+      member: member,
+      astWrapper: AstWrapper.build()
+    })
+    this._pullingAdRepository = new PullingAdRepository()
   }
 
-  /**
-   * Method to define tags.
-   * @param member
-   * @param targetId
-   * @param invCode
-   * @param sizes
-   * @param keywords
-   * @param native
-   */
-  defineTag ({member, targetId, invCode, sizes, keywords, native}) {
-    throw new Error('AppNexusConnector#defineTag must be implemented')
+  display ({id}) {
+    return Promise.resolve()
+      .then(() => this._astClient.showTag({target: id}))
+      .then(null)
   }
 
-  /**
-   * Load tags.
-   */
-  loadTags () {
-    throw new Error('AppNexusConnector#loadTags must be implemented')
+  loadAd ({domElementId, placement, sizes, segmentation, native}) {
+    return Promise.resolve()
+      .then(() => this._astClient
+        .defineTag({
+          member: this._astClient.member,
+          targetId: domElementId,
+          invCode: placement,
+          sizes: sizes,
+          keywords: segmentation,
+          native: native
+        }))
+      .then(astClient => astClient.onEvent({
+        event: AD_AVAILABLE,
+        targetId: domElementId,
+        callback: consumer(this._pullingAdRepository)(domElementId)(AD_AVAILABLE)
+      }))
+      .then(astClient => astClient.onEvent({
+        event: AD_BAD_REQUEST,
+        targetId: domElementId,
+        callback: consumer(this._pullingAdRepository)(domElementId)(AD_BAD_REQUEST)
+      }))
+      .then(astClient => astClient.onEvent({
+        event: AD_ERROR,
+        targetId: domElementId,
+        callback: consumer(this._pullingAdRepository)(domElementId)(AD_ERROR)
+      }))
+      .then(astClient => astClient.onEvent({
+        event: AD_NO_BID,
+        targetId: domElementId,
+        callback: consumer(this._pullingAdRepository)(domElementId)(AD_NO_BID)
+      }))
+      .then(astClient => astClient.onEvent({
+        event: AD_REQUEST_FAILURE,
+        targetId: domElementId,
+        callback: consumer(this._pullingAdRepository)(domElementId)(AD_REQUEST_FAILURE)
+      }))
+      .then(astClient => astClient.loadTags())
+      .then(() => this._pullingAdRepository.find({id: domElementId}))
   }
-
-  /**
-   * Shows tags in the target.
-   * @param target
-   */
-  showTag ({target}) {
-    throw new Error('AppNexusConnector#showTag must be implemented')
-  }
-
-  /**
-   * Resets the state to it's pre uninitialized state.
-   */
-  reset () {
-    throw new Error('AppNexusConnector#clearRequest must be implemented')
-  }
-
-  /**
-   * Refreshes ads on the page.
-   * @param target : an array of ids
-   */
-  refresh (target) {
-    throw new Error('AppNexusConnector#refresh must be implemented')
+  refresh ({id, segmentation}) {
```

Comment:
Could you please add some JSDoc to all methods on this class in order to clarify the meaning of the parameters and the purpose of the method itself? Thanx!

Reply:
I see the requested info is included here: https://github.com/scm-spain/OpenAds-ConnectorAPI/pull/1/files